library/src/main/java/org/xmtp/android/library/ConversationV1.kt

```diff
@@ -0,0 +1,144 @@
+package org.xmtp.android.library
+
+import android.content.res.Resources.NotFoundException
+import kotlinx.coroutines.runBlocking
+import org.xmtp.android.library.codecs.ContentCodec
+import org.xmtp.android.library.codecs.EncodedContent
+import org.xmtp.android.library.codecs.TextCodec
+import org.xmtp.android.library.messages.Envelope
+import org.xmtp.android.library.messages.EnvelopeBuilder
+import org.xmtp.android.library.messages.Message
+import org.xmtp.android.library.messages.MessageBuilder
+import org.xmtp.android.library.messages.MessageV1Builder
+import org.xmtp.android.library.messages.Topic
+import org.xmtp.android.library.messages.decrypt
+import org.xmtp.android.library.messages.header
+import org.xmtp.android.library.messages.sentAt
+import org.xmtp.android.library.messages.toPublicKeyBundle
+import org.xmtp.android.library.messages.walletAddress
+import java.util.Date
+
+data class ConversationV1Container(
+    var peerAddress: String,
+    var sentAt: Date
+) {
+
+    fun decode(client: Client): ConversationV1 =
+        ConversationV1(client = client, peerAddress = peerAddress, sentAt = sentAt)
+}
+
+data class ConversationV1(
+    var client: Client,
+    var peerAddress: String,
+    var sentAt: Date
+) {
+    val topic: Topic
+        get() = Topic.directMessageV1(client.address, peerAddress)
+
+    fun send(content: String, sentAt: Date? = null) {
+        val encoder = TextCodec()
+        val encodedContent = encoder.encode(content = content)
+        send(content = encodedContent)
+    }
+
+    fun <T : Any> send(content: T, options: SendOptions? = null) {
+        val codec = Client.codecRegistry.find(options?.contentType)
+
+        fun <Codec : ContentCodec<T>> encode(codec: Codec, content: Any): EncodedContent {
+            val contentType = content as? T
+            if (contentType != null) {
+                return codec.encode(content = contentType)
+            } else {
+                throw java.lang.NullPointerException()
```

Comment:
Small suggestion: I see us using `IllegalArgumentException` elsewhere with a custom message -- maybe that would be more clear than a generic NPE?